policyenforcer.go

```diff
@@ -46,9 +46,26 @@ type ValidationReport struct {
 	ArtifactReports []*ValidationReport
 }
 
-// PolicyEnforcer is an interface with methods that make policy decisions.
+// Evaluator is an interface that defines methods to aggregate and evaluate
+// verification results generated for an artifact per validation request.
+type Evaluator interface {
+	// VerifyRequired checks if the provided verifier is required to perform
+	// verification on the subject against the artifact. If the verifier is
+	// required, it returns [ErrVerifyRequired].
+	VerifyRequired(ctx context.Context, subjectDigest, artifactDigest string, verifier Verifier) error
```

Comment:
By default, verification is required. In some circumstances, verification is skipped. Precisely, we don't just skip the verification but skip the entire sub-graph. That is, pruning.

Reply:
We can call it `Pruned`?

Reply:
BTW, why is `verifier` required?

Reply:
`verifier` includes the `Verifiable` which can tell if the artifact is a signature of certain type.